modules/control-utility/src/test/java/org/apache/ignite/util/KillCommandsCommandShTest.java

```diff
@@ -195,4 +205,97 @@ public void testCancelUnknownContinuousQuery() {
 
         assertEquals(EXIT_CODE_OK, res);
     }
+
+    /** */
+    @Test
+    public void testCancelConsistencyMissedTask() {
+        int res = execute("--kill", "consistency");
+
+        assertEquals(EXIT_CODE_OK, res);
+    }
+
+    /** */
+    @Test
+    public void testCancelConsistencyTask() throws InterruptedException {
+        String consistencyCancheName = "consistencyCache";
+
+        CacheConfiguration<Integer, Integer> cfg = new CacheConfiguration<>();
+
+        cfg.setName(consistencyCancheName);
+        cfg.setBackups(SERVER_NODE_CNT - 1);
+
+        IgniteCache<Integer, Integer> cache = client.getOrCreateCache(cfg);
+
+        for (int i = 0; i < 10_000; i++)
+            cache.put(i, i);
+
+        CountDownLatch getLatch = new CountDownLatch(SERVER_NODE_CNT); // Each node should send a get request.
+
+        for (IgniteEx server : srvs) {
+            TestRecordingCommunicationSpi spi =
+                ((TestRecordingCommunicationSpi)server.configuration().getCommunicationSpi());
+
+            spi.blockMessages((node, message) -> {
+                if (message instanceof GridNearGetRequest) { // Get request caused by read repair operation.
+                    getLatch.countDown();
+
+                    return true; // Blocking to freeze '--consistency repair' operation.
+                }
+
+                return false;
+            });
+        }
+
+        CountDownLatch thLatch = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                getLatch.await(); // Waiting for consistency repair start.
+            }
+            catch (InterruptedException e) {
+                fail();
+            }
+
+            IgnitePredicate<GridJobWorker> repairJobFilter =
+                worker -> worker.getJob() instanceof VisorConsistencyRepairTask.VisorConsistencyRepairJob;
+
+            for (IgniteEx node : srvs)
+                assertFalse(node.context().job().findJobs(repairJobFilter).isEmpty()); // Found.
+
+            int res = execute("--kill", "consistency");
+
+            assertEquals(EXIT_CODE_OK, res);
+
+            try {
+                assertTrue(GridTestUtils.waitForCondition(() -> {
+                    for (IgniteEx node : srvs)
+                        if (!node.context().job().findJobs(repairJobFilter).isEmpty())
+                            return false;
+
+                    return true;
+                }, 5000L)); // Missed.
+            }
+            catch (IgniteInterruptedCheckedException e) {
+                fail();
+            }
+
+            thLatch.countDown();
+        }).start();
+
+        injectTestSystemOut();
+
+        assertEquals(EXIT_CODE_UNEXPECTED_ERROR, execute("--consistency", "repair", consistencyCancheName, "0"));
+
+        assertContains(log, testOut.toString(), "Operation execution cancelled.");
+        assertContains(log, testOut.toString(), "violations were NOT found [processed=0]");
+
+        thLatch.await();
```

Comment:
Please, use `getTestTimeout()` here to await latch with timeout.

Reply:
This will be interrupted on test interruption if getTestTimeout() is exceeded.